### PR TITLE
compiletest: known-bug / crashes: allow for an "auxiliary" directory to contain files that do not have a "known-bug" directive

### LINKED ISSUE #133009: aux-build `tests/crashes` tests don't seem to work
cc @jieyouxu 

Is it possible to write a `tests/crashes` test with `aux-build`? @camelid tried in #131081 but tidy complains that `crashes/auxiliary/aux123456.rs` doesnt have a `known-bug` annotation/

### PR DESCRIPTION
**Pull request by matthiaskrgr**
Fixes #133009

r? @jieyouxu

### CODE REVIEW

**Comment by jieyouxu** [comment]
FWIW in ui tests the aux build logic supports setup like

- tests/ui/foo/auxiliary/bark.rs
- tests/ui/foo/meow.rs

But if crashes don't support nested auxiliary I guess this logic makes sense for tidy?

**Comment by matthiaskrgr** [comment]
I never really planned for the `crashes` directory to have subdirs .. :sweat_smile:

File: src/tools/tidy/src/known_bug.rs [hunk 1]
```diff
@@ -6,8 +6,22 @@ use crate::walk::*;
...
+            file.iter().map(|s| s.to_string_lossy().into()).collect::<Vec<String>>();
+        let test_path_segments_str =
+            test_path_segments.iter().map(|s| s.as_str()).collect::<Vec<&str>>();
+
+        if !matches!(test_path_segments_str[..], [
+            ..,
+            "tests",
+            "crashes",
+            "auxiliary",
+            _aux_file_rs
```
Comment (jieyouxu) [inline src/tools/tidy/src/known_bug.rs:22]: Question: I think the aux test file check be simplified to something like

```rs
let is_crashes_aux_file = |p| !p.is_symlink()
    && p.is_dir()
    && p.components().rev().nth(1).map(|c| c.as_os_str()).is_some_and(|d| d == "auxiliary");
```

Collecting into `Vec<String>` in `test_path_segments` would do a bunch of small string allocations, right?

**Pull request merged**